### DISCUSSION TITLE
Holochain for distributed DID method

### DISCUSSION THREAD

**Discussion by erlend-sh** [holochain/holochain#4490] [Ideas]
You may have heard of Bluesky and its underlying protocol ATProto lately.

One of the key reasons why bsky is often critiqued as only ‘somewhat decentralized’ is the centralized nature of their `did:plc`– https://web.plc.directory/

Various attempts are now being made to decentralize the bsky did method, either by making did:plc itself more distributed or by coming up with alternative did methods that could potentially replace did:plc as the default.

Here’s an example of an alt method: https://cosign2.witness.co/cid/QmcJGJbcsvpykNMcQ21ryHSGaLkm81gPS52zxtb61dHEtQ

> in short, it's similar to `did:plc` except:
> - DID Method updates are stored on Arweave (for permanence/transparency)
> - updates are timestamped via Witness Protocol (for temporal guarantees at scale)
> 
> this means that it's decentralized (anyone can submit updates, > can't be censored) and scalable (doesn't require a blockchain transaction for each DID update)

The Ethereum dependency however feels out of place to me, although I’m no expert at this stuff.

It just seems a little heavy. And notes like this one sound a tad questionable:

> note: because publishing on the Ethereum Smart Contract is permissionless, it is possible that improper data is included. Any improperly formed data MUST be ignored. Any arweave data blog that *includes* any improperly formatted Witnessed Credentials MUST be ignored entirely.

What makes me most uneasy about Ethereum though is that it’s a financialized network with a myriad of activities entirely unrelated to identity management. I don’t want my identity to be at the mercy of a speculative market with busts and booms. Seems like needless volatility for what ought be a perfectly boring accounting of people’s identities changing every few days/years/decades with minimal network fluctuations.

It’s been quite a while since I last looked at Holochain, but afaik it has always billed itself as ‘much like a blockchain, but not a (wasteful) blockchain’.

I just have this vague intuition that Holochain might be a good fit for a collectively owned/maintained (at least across dozens, maybe thousands) did directory. Curious if this has been explored at all yet.

See also: https://dustycloud.org/blog/how-decentralized-is-bluesky/

**Comment by zippy** [comment]
There has been [work on this in the past](https://github.com/Holo-Host/did-holo-resolver/blob/master/doc/did_holo_spec.md) though we are not running a resolver at this time, and this spec probably needs some updating.  

There is another project [coasys](https://coasys.org/) that is using DIDs along with Holochain.

This is something we really want to get to, timeline unknown however.

**Reply by erlend-sh** [reply]
Came across some other historical tidbits:

- https://github.com/bluesky-social/atproto/issues/810 ..which pointed to 
- A more recent update of the [did:holo spec](https://github.com/WebOfTrustInfo/rwot9-prague/blob/master/draft-documents/did-hc-method.md), which I hadn’t yet connected to the underlying [deepkey](https://github.com/holochain/deepkey) project.

I was reminded of this track upon reading this recent post and ensuing discussion about the weaknesses of the PLC:

[did:plc Considered Harmful](https://merkletr.ee/fast)
https://bsky.app/profile/merkletr.ee/post/3lx35fe5ax22l

It points out a did method that is theoretically backwards compatible with PLC:

https://bsky.app/profile/tentacle.expert/post/3lbizxbtaac2g
https://codeberg.org/xormetric/didfastpoc

That’s the sort of proposal I’m hoping could come out of the Holochain org as well.
